src/main/java/com/exasol/dbbuilder/dialects/exasol/AdapterScript.java

```diff
@@ -53,6 +83,25 @@ public String getContent() {
         return this.content;
     }
 
+    /**
+     * Get if debugging adapter scripts was globally enabled enabled.
+     * 
+     * @return true if debugging adapter scripts was globally enabled
+     */
+    public static boolean isAdapterScriptDebuggingEnabled() {
+        final String debugProperty = System.getProperty("tests.debugAdapterScripts");
```

Comment:
This violates the SoC. An `AdapterScript` object should not know about system properties or if Debugging is enabled. It is a too high level concept for this kind of detail and also not the right place. A system configuration object (can be an immutable singleton in this case) is cleaner.